backend/hardware/devicelist.py

```diff
@@ -8,17 +8,21 @@
 from os.path import exists
 
 def loadHardwareConfiguration():
-  if config.hardwareBoard != "custom":
+  if config.hardwareBoard == "simulation":
+    path = './hardware/boardhardwareconfig/hardware_simulation.yaml'
+    if not exists(path):
+      raise Exception("No simulation board found '{0}'".format(config.hardwareBoard))
+    boardHardware = yaml.safe_load(open(path, 'r'))
+    return { "devices": boardHardware["devices"] }
+  elif config.hardwareBoard != "custom":
```

Comment:
This can be removed since if the value for hardwareBoard in config.py matches the hardware config filename the "if config.hardwareBoard != "custom" case will load the new simulation configuration just fine and nothing custom is needed

Reply:
agree. I was a bit puzzled by this return line as it looks like there is a case to merge two configurations
return { "devices": boardHardware["devices"] + userHardware["devices"]}

thank you for the feedback, I will try incorporate all your suggestions